frontend/src/Quiz/QuizAssign.vue

```diff
@@ -0,0 +1,318 @@
+  return !clazz.assigned || !clazz.deadline || !clazz.duration;
+};
+
+updateGlobalOptions({
+  multiple: false,
+  theme: 'auto',
+  autoClose: 2000,
+  transition: 'slide',
+  dangerouslyHTMLString: true
+});
+
+const isCollapsed = ref<boolean>(true);
+const selectedTeacher = ref<string>(teacher);
+const assignmentsData = ref<AssignmentClass[]>(JSON.parse(assignments) satisfies AssignmentClass[]);
+const teacherNames = ref<string[]>([
+  ...new Set(assignmentsData.value.map((c) => c.teacher)),
+  'All'
+]);
+
+/**
+ * Send assignments to the server
+ */
+const saveAssignments = async () => {
+  const data = await getDataWithCSRF<{
+    message: string;
+    quiz_deletable: boolean;
+    assignments: AssignmentClass[];
+  }>(`/api/quiz/${quiz_id}/assignments`, 'POST', { assignments: assignmentsData.value });
+
+  if (data) {
+    toast.success(data.message);
+    quizDeletable.value = data.quiz_deletable;
+    assignmentsData.value = data.assignments;
+  } else {
+    toast.error('Failed to save assignments');
+  }
+};
+
+/**
+ * Set the same values for all classes
+ * @param target AssignmentClass to be copied from
+ */
+const setToAllClasses = async (target: AssignmentClass) => {
+  const properties = ['assigned', 'deadline', 'duration', 'publish_results'];
+
+  properties.forEach((property) => {
+    assignmentsData.value
+      .filter((c) => c.visible)
+      .forEach((clazz) => {
+        clazz[property] = target[property];
+      });
+  });
+};
+
+/**
+ * Set the same value for all classes by property
+ * @param target AssignmentClass to be copied from
+ * @param property AssignmentClass property to be copied from the target
+ */
+const setToAllClassesByProperty = async (
+  target: AssignmentClass,
+  property: AssignmentClassProperty
+) => {
+  const properties = ['assigned', 'deadline', 'duration'];
+
+  const propertyIndex = properties.indexOf(property);
+
+  assignmentsData.value
+    .filter((c) => c.visible)
+    .forEach((clazz) => {
+      let canAssign = true;
+
+      for (let i = 0; i < propertyIndex; i++) {
+        if (!clazz[properties[i]]) {
+          canAssign = false;
+          break;
+        }
+      }
+
+      if (canAssign) {
+        clazz[property as string] = target[property];
+      }
+    });
+};
+
+/**
+ * Delete assignment values
+ * @param clazz AssignmentClass that values have to be deleted
+ */
+const deleteAssignmentValues = async (clazz: AssignmentClass) => {
+  clazz.assigned = null;
+  clazz.deadline = null;
+  clazz.duration = null;
+  clazz.publish_results = null;
+};
+
+/**
+ * Set publish results for all possible assignments
+ */
+const setPublishToAllClasses = async () => {
+  assignmentsData.value
+    .filter((c) => c.visible)
+    .forEach((clazz) => {
+      if (!isMissingPropertyToAssign(clazz)) {
+        clazz.publish_results = true;
+      }
+    });
+};
+
+/**
+ * Watch for changes in the selected teacher and filter the classes
+ */
+watchEffect(() => {
```

Comment:
This is a really hacky solution :) Please create a `computed` that will return only the visible classes based on `selectedTeacher`, and use that in the template, to avoid modifying the classes.